kafka/src/main/java/org/apache/apex/malhar/kafka/AbstractKafkaInputOperator.java

```diff
@@ -339,6 +394,11 @@ public void assign(Set<AbstractKafkaPartitioner.PartitionMeta> assignment)
     return assignment;
   }
 
+  private boolean isIdempotent()
+  {
+    return windowDataManager != null && !(windowDataManager instanceof WindowDataManager.NoopWindowDataManager);
+  }
```

Comment:
I think HighLevelKafkaConsumer doesn't support emitImmediately. So, this check needs  to be added in isIdempotent() API. Please correct it, if I am wrong.